spi/src/main/java/org/opensearch/ml/common/spi/tools/Tool.java

```diff
@@ -129,5 +132,11 @@ interface Factory<T extends Tool> {
          * @return the default tool version
          */
         String getDefaultVersion();
+
+        /**
+         * Get the default attributes of this tool
+         * @return the default attributes
+         */
+        Map<String, Object> getDefaultAttributes();
```

Comment:
Skills repo also need to implement this method. https://github.com/opensearch-project/skills